backend/src/test/java/com/example/parkingspot/controller/PersonControllerMVCTest.java

```diff
@@ -0,0 +1,137 @@
+package com.example.parkingspot.controller;
+
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import com.example.parkingspot.entity.Car;
+import com.example.parkingspot.entity.Person;
+import com.example.parkingspot.service.CarService;
+import com.example.parkingspot.service.PersonService;
+
+@WebMvcTest(PersonController.class)
+@AutoConfigureMockMvc(addFilters = false)
+// disables security filters
+public class PersonControllerMVCTest {
+
+  @MockBean
+  private PersonService personService;
+  @MockBean
+  private CarService carService;
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Test
+  @DisplayName("Endpoint getPersons - should return 200 OK and Person array in JSON")
```

Comment:
Just a reflection: is there a possibility that an erroneous array in the response JSON for get(all)Persons could get through – i.e. could something more be set as expected than "[ ]" as content? I actually don't know myself though. 😅

Reply:
Nice catch! Didnt think of it at the time that it might aswell return ANY array. Pushing refactor of the test to now check for valid data on a Person obj/entity

Reply:
But then again... the DB might not have any entrys in the table for Person